packages/scenes/src/core/sceneGraph/utils.ts

```diff
@@ -45,6 +45,8 @@ export function cloneSceneObjectState<TState extends SceneObjectState>(
       for (const child of propValue) {
         if (child instanceof SceneObjectBase) {
           newArray.push(child.clone());
+        } else if (typeof child === 'object') {
+          newArray.push({ ...child });
```

Comment:
This is going to be problematic for properties that are `Array`. The individual array values will be spread as properties on the state.

Reply:
Ah, good catch. Do you think something like this would be okay?
```suggestion
        } else if (typeof child === 'object' && !Array.isArray(child)) {
          newArray.push({ ...child });
```